lib/strava/errors/upload_failed_error.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Strava
+  module Errors
+    class UploadFailedError < ::Faraday::ClientError
+      def message
+        response[:body]['error'] || super
```

Comment:
Add a private property called `body` that returns `response[:body]` to avoid repeating it everywhere, and let's either be defensive with `body?['error']` or return `response[:body] || {}` in it, in case something turns out to be `nil`.